dafny 3.0.0

davidcok opened the pull request:
This update to the Dafny formula corresponds to the 3.0.0 release of Dafny. That release uses dotnet 5.0 rather than mono to build and run Dafny.

carlocab commented on Formula/dafny.rb:
```diff
@@ -1,10 +1,10 @@
+  sha256 "5e9af6ca59c3329cd93d792bf9890c55c68c4f656afb19c85d1c44b0c7989fc2"
   license "MIT"
-  revision 3
+  revision 4
```
```suggestion
```
You can remove this now, since this is a version bump.

davidcok replied:
Done

carlocab commented:
```
==> make exe runtime
(cd /private/tmp/dafny-20210204-58618-qp6xzp/dafny-3.0.0/ ; dotnet build Source/Dafny.sln ) ## includes parser
(cd /private/tmp/dafny-20210204-58618-qp6xzp/dafny-3.0.0//Source/DafnyRuntime/DafnyRuntimeJava; ./gradlew -q clean copyJarToBinaries)
No Java runtime present, requesting install.
```
Looks like you are missing some dependencies?

carlocab commented:
Same error message. You're still missing dependencies.

davidcok commented:
I have added them now — but I would appreciate assistance on all the audit messages that now appear, that did not before — regarding packages I do not reference and and the format of sha256 lines

> On Feb 4, 2021, at 9:55 AM, Carlo Cabrera <notifications@github.com> wrote:
> 
> 
> Same error message. You're still missing dependencies.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Homebrew/homebrew-core/pull/70359#issuecomment-773369519>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABITDQHXGG7R7TZOTLO2AI3S5KYM5ANCNFSM4XB65APQ>.
>

carlocab commented:
>packages I do not reference

That's not an audit error; that's the build failing because it was looking for a non-existent Java installation.

>format of sha256 lines

This was a minor `brew` hiccup; you can ignore that.

carlocab commented on Formula/dafny.rb:
```diff
@@ -18,31 +17,22 @@ class Dafny < Formula
-        revision: "9e74c3271f430adb958908400c6f6fce5b59000a"
-  end
+  depends_on "dotnet"
+  depends_on "gradle"
+  depends_on "openjdk"
```
Are these needed at runtime?

davidcok commented:
Changed grade to build only. Adjusted order per the audit.

> On Feb 4, 2021, at 10:08 AM, Carlo Cabrera <notifications@github.com> wrote:
> 
> 
> @carlocab commented on this pull request.
> 
> In Formula/dafny.rb <https://github.com/Homebrew/homebrew-core/pull/70359#discussion_r570298577>:
> 
> > +  depends_on "gradle"
> +  depends_on "openjdk"
> Are these needed at runtime?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Homebrew/homebrew-core/pull/70359#pullrequestreview-583491962>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABITDQEE4DAZZ5ZTM62V5PLS5K2AJANCNFSM4XB65APQ>.
>

carlocab commented:
You forgot to push your changes.

carlocab commented:
You're proposing changes to an outdated version of the formula. Please rebase this PR against master then push your changes again. If you need help with that, let me know.

carlocab commented:
That just made it worse, because your local master branch is out of date.

Anyway, I've fixed it now, I think. Please make sure your local copy of the repository is up to date before pushing changes.

Also, please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit messages. Refer to the [documentation](https://docs.brew.sh/Formula-Cookbook#commit) for more details.
